### PR TITLE
Llama 2 on the Hub

### PR DESCRIPTION
**Pull request by osanseviero**

### CODE REVIEW

File: llama2.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+## Why Llama 2?
+
+The Llama 2 release introduces a family of pretrained and fine-tuned LLMs, ranging in scale from 7B to 70B parameters (7B, 13B, 70B). The pretrained models come with significant improvements over the Llama 1 models, including being trained on 40% more tokens, having a much longer context length (4k tokens 🤯), and using grouped-query attention for fast inference of the 70B model🔥!
+
+However, the most exciting part of this release is the fine-tuned models (Llama 2-Chat), which have been optimized for dialogue applications using [Reinforcement Learning from Human Feedback (RLHF)](https://huggingface.co/blog/rlhf). Across a wide range of helpfulness and safety benchmarks, the Llama 2-Chat models perform better than most open models and achieve comparable performance to ChatGPT according to human evaluations.
+
+| ![mqa](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/llama-rlhf.png) |
+
+*****************************TODO: add paper link to an image*****************************
```
Comment (philschmid) [inline llama2.md:48]: we can use: https://scontent-fra3-2.xx.fbcdn.net/v/t39.2365-6/10000000_6495670187160042_4742060979571156424_n.pdf?_nc_cat=104&ccb=1-7&_nc_sid=3c67a6&_nc_ohc=GK8Rh1tm_4IAX8b5yo4&_nc_ht=scontent-fra3-2.xx&oh=00_AfDtg_PRrV6tpy9UmiikeMRuQgk6Rej7bCPOkXZQVmUKAg&oe=64BBD830

Comment (philschmid) [inline llama2.md:48]:
```suggestion
_image from [Llama 2: Open Foundation and Fine-Tuned Chat Models](https://scontent-fra3-2.xx.fbcdn.net/v/t39.2365-6/10000000_6495670187160042_4742060979571156424_n.pdf?_nc_cat=104&ccb=1-7&_nc_sid=3c67a6&_nc_ohc=GK8Rh1tm_4IAX8b5yo4&_nc_ht=scontent-fra3-2.xx&oh=00_AfDtg_PRrV6tpy9UmiikeMRuQgk6Rej7bCPOkXZQVmUKAg&oe=64BBD830)_
```

from we can use:

File: llama2.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+*we’re currently running evaluation of the Llama 2 70B (non chatty version). This table will be updated with the results.
+
+
+## Demo
+
+You can easily try the Big Llama 2 Model (70 billion parameters!) in [this Space](https://huggingface.co/spaces/ysharma/Explore_llamav2_with_TGI) or in the playground embedded below:
+
+<script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/3.37.0/gradio.js"> </script>
+<gradio-app space="ysharma/Explore_llamav2_with_TGI"></gradio-app>
```
Comment (philschmid) [inline llama2.md:76]: will this be the final space?

File: llama2.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+In the following code snippet, we show how to run inference with transformers
+
+
+# TODO: should we change the example to use system prompt/etc?
+```python
+from transformers import AutoTokenizer
+import transformers
+import torch
+
```
Comment (philschmid) [inline llama2.md:102]: should we add something for login or the token? otherwise it will fail?

File: llama2.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+**[Text Generation Inference](https://github.com/huggingface/text-generation-inference)** is a production-ready inference container developed by Hugging Face to enable easy deployment of large language models. It has features such as continuous batching, token streaming, tensor parallelism for fast inference on multiple GPUs, and production-ready logging and tracing. 
+
+You can try out Text Generation Inference on your own infrastructure, or you can use Hugging Face's **[Inference Endpoints](https://huggingface.co/inference-endpoints)**. To deploy a Llama 2 model, go to the **[model page](https://huggingface.co/tiiuae/falcon-7b-instruct)** and click on the **[Deploy -> Inference Endpoints](https://ui.endpoints.huggingface.co/new?repository=tiiuae/falcon-7b-instruct)** widget.
+
+- For 7B models, we advise you to select "GPU [medium] - 1x Nvidia A10G".
+- For 13B models, we advise you to select "GPU [xlarge] - 1x Nvidia A100".
+- For 70B models, we advise you to select "GPU [xxlarge] - 8x Nvidia A100".
+
+*Note: You might need to request a quota upgrade via email to **[api-enterprise@huggingface.co](mailto:api-enterprise@huggingface.co)** to access A100s*
```
Comment (philschmid) [inline llama2.md:146]: fixes _italic_

```suggestion
_Note: You might need to request a quota upgrade via email to **[api-enterprise@huggingface.co](mailto:api-enterprise@huggingface.co)** to access A100s_
```

File: llama2.md (hunk 2 above)
Comment (osanseviero) [reply llama2.md:76]: I think so, as there is no other Space at the moment and it's the only place to play with the 70B model. We can update if wanted

File: llama2.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+```
+Result: I liked "Breaking Bad" and "Band of Brothers". Do you have any recommendations of other shows I might like?
+Answer:
+Of course! If you enjoyed "Breaking Bad" and "Band of Brothers," here are some other TV shows you might enjoy:
+1. "The Sopranos" - This HBO series is a crime drama that explores the life of a New Jersey mob boss, Tony Soprano, as he navigates the criminal underworld and deals with personal and family issues.
+2. "The Wire" - This HBO series is a gritty and realistic portrayal of the drug trade in Baltimore, exploring the impact of drugs on individuals, communities, and the criminal justice system.
+3. "Mad Men" - Set in the 1960s, this AMC series follows the lives of advertising executives on Madison Avenue, expl
+```
+
+And although the model has *only* 4k tokens of context, you can use techniques supported in `transformers` such as rotary position embeddings ([tweet](https://twitter.com/joao_gante/status/1679775399172251648)) to push it further!
```
Comment (gante) [inline llama2.md:140]:
```suggestion
And although the model has *only* 4k tokens of context, you can use techniques supported in `transformers` such as rotary position embedding scaling ([tweet](https://twitter.com/joao_gante/status/1679775399172251648)) to push it further!
```
("rotary position embedding" is the name of the position embedding that is part of the model, "rotary position embedding scaling" is the name of the thing that enables us to go beyond the original sequence length)

**Review by gante — COMMENTED** [review]
nit :)

**Comment by gante** [comment]
I'll open a PR!

**Pull request merged**